### ISSUE TITLE
Large Shadow phantom braking

### ISSUE THREAD

**Issue by 8a9phrywet978ui** [commaai/openpilot#20704]
**Describe the bug**

Tall vehicles when casting a shadow into the adjacent lane cause openpilot to mis-identify the shadows as vehicles, and abruptly brake even with no actual vehicle in front of you in your lane

**How to reproduce or log data**

Very Large Vehicle or Semi in Right Lane, openpilot vehicle in Left Lane,  sun casting shadow of the Semi onto majority of left lane 

**Expected behavior**

Not doing this

**Device/Version information (please complete the following information):**
 - Device: Comma 2
 - Dongle ID: 9d8c378664ae56f2
 - Route: No Longer Have, multiple instances on April 12th throughout my 14 hour drive.
 - Timestamp: N/A, after Noon
 - Version: Release2-staging
 - Car make/model: 2019 Toyota Corolla Hatchback

**Additional context**

Other users reporting this issue on the discord, hopefully one of them can provide a better example

**Comment by pd0wm** [comment]
Without a route this is pretty hard to look into for us. Let us know when you have an example.

**Comment by ShaneSmiskol** [comment]
I've noticed similar behavior on the latest KL model. It's super eager at picking out leads, when coming up on stopped traffic in multiple lanes it will quickly switch between the adjacent lanes until I get close enough. This also happens when there's no traffic in my lane, so it will brake for a moment until it realizes there's no lead.

**Comment by adeebshihadeh** [comment]
Closing this due to inactivity. Let us know if you have a route for us to look into.

**Comment by Achilles308** [comment]
> Closing this due to inactivity. Let us know if you have a route for us to look into.

I reported a shadow bridge braking example on discord some 4 weeks or so ago. I tried to find the bridge that it happened at, but with so many to pick from and no readily easily available brake logs on cabana, I was not able to find it. It broke from 70 to 65... so minor... but still a braking event for no reason.

This even was a braking event after a street light and into the shadow behind it (best I can tell, that is what I think it was braking for at least). The set speed was 40 and sometime around the streetlight or the shadow, OP broke to 35.

comma 3
master-ci 0.8.10
dfb993cc43ba69dc|2021-09-09--21-50-02--18 and 19
22:08::57 Local time
2020 Toyota Corolla Hatchback

Please reopen this, as it does happen from time to time and I'd like to log them better. It would be helpful to know if this is a VOACC or a radar issue?

**Comment by stwendl** [comment]
I can confirm this too on latest version on c3x on a maverick. But it seems to occur only when experimental features are enabled. I noticed sudden breaks when no car is in front of me but there were shadows thrown by trees on the side of the road and I just passed a cul de sac with no vehicles waiting. Seems the optical longitudinal control gets confused by shadows but not when using fords radar sensing. Perhaps a combination, if possible, would be the solution. I haven’t used experimental features as they are too experimental. What also would be nice is to be able to enable individual experimental features instead of the whole lot. I think other forks have that

**Comment by stwendl** [comment]
Update, I recently upgraded to a comma4 and no longer experience those sudden breaks when getting close to shadows. Perhaps 0.10.3 has no longer the issue or the season had something to do.  I noticed this on my c3x, which I no longer have, around fall of 25.  Due to other unexpected behavior, I did not use the experimental features often but tried once in a while. The c4 seems to not show the sudden breaking in the same spot.  

However I noticed some odd behavior when driving over a particular spot on a road where the c4 seems to loose control. I think it has something to do with a surface repair optical confusion. I think I can make a video available. Symptoms are loss of lane centering for a second or two.
